### PR TITLE
Overhauls large artifact harvesting

### PR DESCRIPTION
**Pull request by BobdaBiscuit**
The aim of this PR is to make working with anomaly batteries more robust so that secondary effects can be extracted, without any methods to cheese the system.

Current behaviour:
- Put artifact on the scanner
- Insert a battery
- Harvest
- The primary effect of that artifact magically goes into the battery, even if you don't know what it does

PR behaviour:
- Put artifact on the scanner and initialise the containment field
- Use the trigger of whatever effect you want in order to toggle the artifact
- Isolate the effect you want
- Insert yer battery
- Harvest
- The effect you selected is put into the battery, you have to have discovered it's trigger in order to do this

[Video Demo](https://www.youtube.com/watch?v=ykS6QEEpt8I)
Artifact was non togglable teleport on touch, and togglable radiation on heavy brute.

TODO:
- [x] Fix ontouch effects (harm etc) bypassing the containment field and being impossible to isolate.

Possible Features:
- [x] Instead of disallowing the isolation of effects when both are activated, allow the user to select one to continue with

:cl:
* tweak: The Large Artifact effect extractor can now extract secondary effects. The artifact must be inside the containment field with the effects you want to extract active, you can then isolate them and pick what one to put into a battery. The containment field as is blocks all artifact effects apart from darkness and forcefield.
* rscadd: Large Artifact secondary effects now show a message when toggled.

### CODE REVIEW

**Comment by BobdaBiscuit** [comment]
Tested everything sane I could, it all works.
Only real issue I've found is some effects can bypass the containment field as they don't use process(), ie forcefield. It shouldn't be a major issue.

**Comment by AI-Is-Rogue** [comment]
So will this fix some artifact effects not doing anything in conjunction with the batteries? 
edit: And will this work with EMP artifacts? Because I was unable to harvest those when they were activated, due to it fucking up all the systems.

**Comment by BobdaBiscuit** [comment]
Due to PRs having to be atomic I haven't touched that, however it is on my shitlist of things to do right after if this gets merged.

What it does do is makes secondary effects able to be harvested, which was previously impossible.

edit: In regards to emp artifacts the containment field currently stops every effect type that uses process() for it's effects, which includes EMP.

**Comment by JMWTurner** [comment]
Oh fuck yes

**Comment by BobdaBiscuit** [comment]
On touch effects (eg harm, gas) are now blocked by the containment field.
Since they cannot be toggled, you have 10 seconds after triggering it to isolate the effect.

~~Working on blocking effects such as forcefield and darkness.~~

File: code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm]: `10 SECONDS`

File: code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm]: `10 SECONDS`

File: code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm]: `10 SECONDS`

File: code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm]: `10 SECONDS`

File: code/modules/research/xenoarchaeology/artifact/effect.dm
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/effect.dm]: Why is this not inside the `visible_message`?

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Review by PJB3005 — APPROVED** [review]

File: code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm
Comment (N3X15) [inline code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm:88]: That argument smells like a magic number.  Given that the current proc signature is `/datum/artifact_effect/proc/ToggleActivate(var/reveal_toggle = 1)`, this should probably be made a `#define` so it's clearer what the difference between 0 and 2 are.

Comment (BobdaBiscuit) [reply code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm:88]: Can I define it there? The idea is to differentiate between secondary and primary effects toggling, hence the numbers.

Comment (N3X15) [reply code/modules/research/xenoarchaeology/artifact/artifact_unknown.dm:88]: Nevermind then.

**Pull request merged**
